packages/hardhat-plugin-ethers/src/index.ts

```diff
@@ -0,0 +1,24 @@
+import "@nomicfoundation/hardhat-ignition";
+import "@nomicfoundation/hardhat-ethers";
+import { extendEnvironment } from "hardhat/config";
+import { HardhatPluginError, lazyObject } from "hardhat/plugins";
+
+import "./type-extensions";
+
+/**
+ * Add an `ignition` object to the HRE.
+ */
+extendEnvironment((hre) => {
+  if (hre.ignition !== undefined && hre.ignition.type !== "ethers") {
```

Comment:
Should we perhaps assign the respective `IgnitionHelper` instances each to their own unique properties on the `hre` object so they could be used simultaneously?

Reply:
I think you have right with the check and enforcement of only one of either viem or ethers.